src/usr/share/gpgmailer/gpgmailer.py

```diff
@@ -15,156 +13,164 @@
+        self.mailsender = mailsender.MailSender(self.config)
+
+        self.outbox_path = os.path.join(self.config['watch_dir'], 'outbox')
+
+        # Set expiration_warning_message so that it is not detected as changed immediately
+        #   after gpgmailer-daemon sends an email.
+        self.expiration_warning_message = gpgkeyverifier.get_expiration_warning_message(time.time())
 
+        self.logger.info('Done initializing gpgmailer module.')
+
+    # Gpgmailer's main loop. Reads the watch directory and then calls other modules
+    #   to build and send email.
+    def start_monitoring(self):
         try:
-            self.smtp.sendmail(self.config['sender']['email'], recipients, encrypted_message_string)
+            while True:
+                loop_start_time = time.time()
+
+                self.recipients = self.gpgkeyverifier.get_valid_recipients(loop_start_time)
+                self.keys = self.gpgkeyverifier.get_valid_keys(loop_start_time)
+
+                new_expiration_warning_message = self.gpgkeyverifier.get_expiration_warning_message(loop_start_time)
+                self._update_expiration_warning_message(new_expiration_warning_message)
+                self._send_email_if_new_expiration_warning_messages()
+
+                for file_name in self._get_file_list():
+                    self.logger.info("Found queued email in file %s." % file_name)
+                    message_dict = self._read_message_file(file_name)
+
+
+                    # Set default subject if the queued message does not have one.
+                    if message_dict['subject'] == None:
+                        message_dict['subject'] = self.config['default_subject']
+
+                    encrypted_message = self._build_encrypted_message(loop_start_time, message_dict)
+
+                    self.mailsender.sendmail(message_string=encrypted_message, recipients=self.recipients)
+                    self.logger.info('Message %s sent successfully.' % file_name)
+
+                    os.remove(os.path.join(self.outbox_path, file_name))
+
+                time.sleep(self.config['main_loop_delay'])
+
+        except gpgkeyverifier.NoUsableKeysException as e:
+            self.logger.critical('No keys available for encryption. Exiting.')
+            sys.exit(1)
+        except gpgkeyverifier.SenderKeyExpiredException as e:
+            self.logger.critical('Sender key has expired and sending unsigned emails is not allowed. Exiting.')
+            sys.exit(1)
         except Exception as e:
-            self.logger.error("Failed to send: %s: %s\n" % (type(e).__name__, e.message))
-            self.logger.debug(traceback.format_exc())
+            self.logger.error('Exception %s:%s.' % (type(e).__name__, e.message))
+            self.logger.error(traceback.format_exc())
+
+    # Return a list of non-directory files in the outbox directory.
+    def _get_file_list(self):
```

Comment:
If you would never use it again, then maybe it's better to add a comment than to abstract it away as a method. Don't think it's super important.